### PR TITLE
fix(a2a): use protocol-standard contextId as sessionId instead of req…

### LINKED ISSUE #1603: [Bug]:A2A在传递userId、sessionId时发送逻辑和取值逻辑不一致，导致userId、sessionId在server端丢失。
版本为1.1.0-RC2

在client端调用agent发送请求时，会调用A2aAgent的doCall方法，doCall方法会对消息进行处理，将多条消息合并为一条消息。

Image

convertFromMsg方法会将消息的metadata变为<msgId,metadata>的键值对。

Image

但是在server端进行解析时存在不一致，导致下面两个方法无法正确获取到userId、sessionId

Image

Image

这是我的client端代码
public class A2aAgentClient {

    public static void main(String[] args) {
        // 创建 A2A Agent
        A2aAgent agent = A2aAgent.builder()
                .name("remote-agent")
                .agentCardResolver(
                        WellKnownAgentCardResolver.builder()
                                .baseUrl("http://127.0.0.1:8946")
                                .build()
                )
                .build();
        Msg response = agent.call(Msg.builder().textContent("你好 我是张三").metadata(Map.of("userId","11")).build())
                .block();
        String collect = response.getContent()
                .stream()
                .filter(TextBlock.class::isInstance)
                .map(TextBlock.class::cast)
                .map(TextBlock::getText)
                .collect(Collectors.joining());
        System.out.println(collect);
    }
}
这是server端debug的截图，可以明显看到metadata中的结构

Image

### PR DESCRIPTION
**Pull request by pymBupt**
…uiring metadata hardcoded key

Previously, AgentScopeAgentExecutor.getSessionId() read sessionId exclusively from message.metadata["sessionId"], forcing all A2A clients to hardcode this custom metadata key — inappropriate for multi-client interoperability.

The A2A protocol defines Message.contextId as the standard field for session/conversation context tracking.

Changes:
- Prefer contextId (via RequestContext.getContextId()) as the sessionId source
- Fallback to metadata["sessionId"] for backward compatibility
- Add 3 unit tests covering the resolution strategy

Resolves #1603
Related: #820

## AgentScope-Java Version

[The version of AgentScope-Java you are working on, e.g. 1.0.12, check your pom.xml dependency version or run `mvn dependency:tree | grep agentscope-parent:pom`(only mac/linux)]

## Description

[Please describe the background, purpose, changes made, and how to test this PR]

## Checklist

Please check the following items before code is ready to be reviewed.

- [ ]  Code has been formatted with `mvn spotless:apply`
- [ ]  All tests are passing (`mvn test`)
- [ ]  Javadoc comments are complete and follow project conventions
- [ ]  Related documentation has been updated (e.g. links, examples, etc.)
- [ ]  Code is ready for review

### CODE REVIEW

File: agentscope-extensions/agentscope-extensions-protocol/agentscope-extensions-a2a/agentscope-extensions-a2a-server/src/test/java/io/agentscope/core/a2a/server/executor/AgentScopeAgentExecutorTest.java [hunk 1]
```diff
@@ -551,6 +551,150 @@ void testHandleExceptionDuringTaskCancellation() throws JSONRPCError {
         }
     }
 
+    @Nested
+    @DisplayName("Session ID Resolution Tests")
+    class SessionIdResolutionTests {
+
+        @Test
+        @DisplayName("Should use contextId as sessionId when contextId is present")
+        void testSessionIdFromContextId() throws JSONRPCError {
```
Comment (AgentScopeJavaBot) [inline agentscope-extensions/agentscope-extensions-protocol/agentscope-extensions-a2a/agentscope-extensions-a2a-server/src/test/java/io/agentscope/core/a2a/server/executor/AgentScopeAgentExecutorTest.java:560]: [recommended] **Test coverage gap**: The existing 3 tests cover contextId present, metadata fallback, and both present. Consider adding a test for the edge case where `context.getContextId()` returns `null` (not empty string) and no metadata sessionId exists — verifying the method returns empty string. This path is handled correctly in the code (`contextId != null` check) but has no test.

Comment (AgentScopeJavaBot) [inline agentscope-extensions/agentscope-extensions-protocol/agentscope-extensions-a2a/agentscope-extensions-a2a-server/src/test/java/io/agentscope/core/a2a/server/executor/AgentScopeAgentExecutorTest.java:560]: [nitpick] **Test code duplication**: The three test cases share extensive mock setup boilerplate (`MessageSendParams` mock, `AtomicReference<AgentRequestOptions>` capture, `mockAgentRunner.stream` stubbing, `executor.execute` call). Consider extracting a private helper method like `executeAndCaptureOptions(contextId, metadataMap)` to reduce duplication.

**Review by AgentScopeJavaBot — APPROVED** [review]
🤖 AI Review

This PR changes `AgentScopeAgentExecutor.getSessionId()` to prefer the A2A protocol-standard `Message.contextId` field over the custom `metadata["sessionId"]` key, while maintaining backward compatibility via fallback. This correctly addresses issue #1603 where all A2A clients were forced to hardcode a custom metadata key. The change is minimal (~24 lines in production code), backward-compatible, and well-documented with Javadoc. Three new unit tests cover the priority resolution strategy. The direction is correct and the implementation is clean. Minor suggestion: add test coverage for the edge case where `context.getContextId()` returns `null`.

File: agentscope-extensions/agentscope-extensions-protocol/agentscope-extensions-a2a/agentscope-extensions-a2a-server/src/test/java/io/agentscope/core/a2a/server/executor/AgentScopeAgentExecutorTest.java (hunk 1 above)
Comment (AgentScopeJavaBot) [inline agentscope-extensions/agentscope-extensions-protocol/agentscope-extensions-a2a/agentscope-extensions-a2a-server/src/test/java/io/agentscope/core/a2a/server/executor/AgentScopeAgentExecutorTest.java:560]: [recommended] **Test coverage gap**: The existing 3 tests cover contextId present, metadata fallback, and both present. Consider adding a test for the edge case where `context.getContextId()` returns `null` (not empty string) and no metadata sessionId exists — verifying the method returns empty string. This path is handled correctly in the code (`contextId != null` check) but has no test.

Comment (AgentScopeJavaBot) [inline agentscope-extensions/agentscope-extensions-protocol/agentscope-extensions-a2a/agentscope-extensions-a2a-server/src/test/java/io/agentscope/core/a2a/server/executor/AgentScopeAgentExecutorTest.java:560]: [nitpick] **Test code duplication**: The three test cases share extensive mock setup boilerplate (`MessageSendParams` mock, `AtomicReference<AgentRequestOptions>` capture, `mockAgentRunner.stream` stubbing, `executor.execute` call). Consider extracting a private helper method like `executeAndCaptureOptions(contextId, metadataMap)` to reduce duplication.

**Review by AgentScopeJavaBot — APPROVED** [review]
🤖 AI Review

This PR changes `AgentScopeAgentExecutor.getSessionId()` to prefer the A2A protocol-standard `Message.contextId` field over the custom `metadata["sessionId"]` key, while maintaining backward compatibility via fallback. This correctly addresses issue #1603 where all A2A clients were forced to hardcode a custom metadata key. The change is minimal (~24 lines in production code), backward-compatible, and well-documented with Javadoc. Three new unit tests cover the priority resolution strategy. The direction is correct and the implementation is clean. Minor suggestion: add test coverage for the edge case where `context.getContextId()` returns `null`.

File: agentscope-extensions/agentscope-extensions-protocol/agentscope-extensions-a2a/agentscope-extensions-a2a-server/src/main/java/io/agentscope/core/a2a/server/executor/AgentScopeAgentExecutor.java [hunk 2]
```diff
@@ -139,7 +139,29 @@ private String getUserId(Message message) {
...
+     *
+     * @param context the request context containing the resolved contextId
+     * @param message the original A2A message
+     * @return the sessionId, or empty string if not found
+     */
+    private String getSessionId(RequestContext context, Message message) {
+        // Prefer the protocol-standard contextId field
+        String contextId = context.getContextId();
+        if (contextId != null && !contextId.isEmpty()) {
+            return contextId;
```
Comment (oss-maintainer) [inline agentscope-extensions/agentscope-extensions-protocol/agentscope-extensions-a2a/agentscope-extensions-a2a-server/src/main/java/io/agentscope/core/a2a/server/executor/AgentScopeAgentExecutor.java:162]: **[Warning]** This is a silent session-identity change for existing clients. Before, `sessionId` was exactly `metadata["sessionId"]`; after, a request that carries *both* a `contextId` and `metadata["sessionId"]` resolves to the `contextId`, so every store keyed by session (agent state persistence, memory, per-session isolation in distribution) starts a brand-new session on upgrade for anyone already passing a custom key. Precedence order is a product decision, but the compatible one is usually explicit-intent-first (`metadata` when present, `contextId` otherwise), or an opt-in config flag; please also call it out in the A2A docs/release notes if the flip is intentional.

**Review by oss-maintainer — COMMENTED** [review]
### Summary
Switches the A2A server to resolve the agent sessionId from the protocol-standard `contextId`, falling back to the legacy `metadata["sessionId"]`. The direction is right — standard A2A clients should not have to invent a metadata key — but the precedence order silently re-keys sessions for existing clients, and the mocked tests do not pin down what `RequestContext.getContextId()` actually holds when the client omits it.

Note: the `license/cla` check is still `pending`, so this review is a COMMENT only — please sign the CLA (see reminder) before a maintainer looks at merging.

---
*Automated review by github-manager-bot*

File: agentscope-extensions/agentscope-extensions-protocol/agentscope-extensions-a2a/agentscope-extensions-a2a-server/src/main/java/io/agentscope/core/a2a/server/executor/AgentScopeAgentExecutor.java [hunk 3]
```diff
@@ -139,7 +139,29 @@ private String getUserId(Message message) {
...
+     * can be correctly routed to per-session state without requiring a custom metadata key.
+     *
+     * @param context the request context containing the resolved contextId
+     * @param message the original A2A message
+     * @return the sessionId, or empty string if not found
+     */
+    private String getSessionId(RequestContext context, Message message) {
+        // Prefer the protocol-standard contextId field
+        String contextId = context.getContextId();
+        if (contextId != null && !contextId.isEmpty()) {
```
Comment (oss-maintainer) [inline agentscope-extensions/agentscope-extensions-protocol/agentscope-extensions-a2a/agentscope-extensions-a2a-server/src/main/java/io/agentscope/core/a2a/server/executor/AgentScopeAgentExecutor.java:161]: **[Warning]** Please confirm what `RequestContext.getContextId()` returns for a `message/send` where the client sent *no* `contextId`. If the SDK materialises a generated contextId for the new task before `execute()` runs, then `contextId != null && !isEmpty()` is true for every fresh conversation and clients that relied on `metadata["sessionId"]` for continuity lose it — the opposite of the intent. The three new tests all stub `getContextId()` to a literal, so this path is exactly the one they cannot catch. An extra test with `getContextId() -> null` plus non-empty `metadata.sessionId` (and, if the SDK does generate one, a note in the javadoc) would settle it.

File: agentscope-extensions/agentscope-extensions-protocol/agentscope-extensions-a2a/agentscope-extensions-a2a-server/src/main/java/io/agentscope/core/a2a/server/executor/AgentScopeAgentExecutor.java [hunk 4]
```diff
@@ -139,7 +139,29 @@ private String getUserId(Message message) {
...
+    /**
+     * Extract sessionId from the A2A request.
+     *
+     * <p>Resolution order:
+     * <ol>
+     *     <li>A2A protocol standard field: {@code Message.contextId} (via {@code RequestContext.getContextId()}).</li>
+     *     <li>Backward-compatible fallback: {@code Message.metadata["sessionId"]}.</li>
+     * </ol>
+     *
+     * <p>This ensures any standard A2A client that sets {@code contextId} on the message
```
Comment (oss-maintainer) [inline agentscope-extensions/agentscope-extensions-protocol/agentscope-extensions-a2a/agentscope-extensions-a2a-server/src/main/java/io/agentscope/core/a2a/server/executor/AgentScopeAgentExecutor.java:151]: **[Info]** The javadoc describes the `metadata` branch as a "Backward-compatible fallback", but the combination "contextId always wins" is not backward compatible for requests that carry both (see the comment on the `return contextId;` line). Worth aligning the wording with the actual contract, and stating which of the two IDs is the durable session key for state/memory so downstream readers do not have to infer it.

**Comment by oss-maintainer** [comment]
**CLA Not Signed**

The Contributor License Agreement (CLA) check is currently pending on this PR (`license/cla`: Contributor License Agreement is not signed yet.). This PR **cannot be merged** until the CLA is signed.

@pymBupt please sign the CLA via the CLA assistant badge in the comment above, or visit https://cla-assistant.io/agentscope-ai/agentscope-java. Once signed, the `license/cla` status will turn green.

---
*Automated check by github-manager-bot*
